cli/core/pkg/artifact/local.go

```diff
@@ -45,3 +45,27 @@ func (l *LocalArtifact) Fetch() ([]byte, error) {
 	}
 	return b, nil
 }
+
+// FetchTest reads test plugin artifact based on the local plugin artifact path
+// To fetch the test binary from the plugin we are using plugin binary path and creating test plugin path from it
+// If the plugin binary path is `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/tanzu-cluster-darwin_amd64`
+// then test plugin binary will be under `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/test/` directory
```

Comment:
q: Will the binary will be under `artifaces/darwin/amd64/cli/cluster/vX.Y.Z-dev/test` directory for any future versions of framework?

Reply:
`artifaces/darwin/amd64/cli` is just the directory that we pass with `--artifacts` flag with `tanzu builder cli compile` command. If we need to change the directory we pass with `--artifacts` flag this might change the root level directory but assumption here is on how builder plugin places plugin and test plugin binaries. Assuming we are passing ${ARTIFACTS_DIR} as part of `--artifacts` flag with compile command...
```
tanzu builder cli compile --path ./cmd/cli/plugin --artifacts ${ARTIFACTS_DIR}
```
the plugin binary will be under `${ARTIFACTS_DIR}/<plugin-name>/tanzu-<plugin-name>-darwin_amd64` and test plugin binary will be under `${ARTIFACTS_DIR}/<plugin-name>/test/` directory.